### PR TITLE
Fixed #1613 and #1588. Add EOL, scientific notiation, and adjust whitespace support

### LINKED ISSUE #1613: Lua Beautifier replaces CRLF with LF
# Description

CRLF is replaced with LF file endings after beautifying,   
Tested with Markdown and LUA beautifiers and they are both affected.
Tested with JS and it was not affected

Tested with Line ending package settings to both system and CRLF

# Input Before Beautification

File has CRLF endings

# Expected Output

File keeps same line endings 

# Actual Output

File has LF endings

# Steps to Reproduce

Open an lua file with CRLF line endings
Beutify

# Debug

Here is a link to the `debug.md` Gist: https://gist.github.com/8ce276ad58d4ede7736c292ac973b0ae

# Checklist

I have:
- [ ] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [x] Filled out the Input, Expected, and Actual sections above or have edited/removed them in a way that fully describes the issue.
- [x] Generated debugging information by executing `Atom Beautify: Help Debug Editor` command in Atom and added link for `debug.md` Gist to this issue

### PR DESCRIPTION
**Pull request by hxsf**
### What does this implement/fix? Explain your changes.
- add scientific notation support
- add EOL support
- add option to control whether adjust whitespace

### Does this close any currently open issues?
#1613 and #1588

### Checklist

Check all those that are applicable and complete.

- [x] Merged with latest `master` branch
- [x] Regenerate documentation with `npm run docs`
- [ ] Update `CHANGELOG.md`
- [x] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

### CODE REVIEW

File: src/languages/lua.coffee [hunk 1]
```diff
@@ -19,4 +19,9 @@ module.exports = {
 
   defaultBeautifier: "Lua beautifier"
 
+  options:
+    do_not_change_whitespace:
```
Comment (Glavin001) [inline src/languages/lua.coffee]: I do not like the name of this option. Is there a clearer way to display this to the users?

Do we even need it? I think what `adjust_space` is doing is adding/removing whitespace. Is that not something we want the beautifier to always do?

File: src/languages/lua.coffee [hunk 2]
```diff
@@ -19,4 +19,9 @@ module.exports = {
...
+      type: 'boolean'
+      default: false
+      description: "do not adjust whitespace"
```
Comment (Glavin001) [inline src/languages/lua.coffee]: - [ ] Capitalize `do` (to `Do`) so it does not look unprofessional
- [ ] Consider improving the description. What does "adjust whitespace" mean? Why would a user not want to do it?

File: src/languages/lua.coffee (hunk 1 above)
Comment (hxsf) [inline src/languages/lua.coffee]: yes, I agree with that.
maybe someone like use `fn(arg1)` and other people like `fn( arg1 )`

It just give users  a choose to only format indent, but don't adjust their whitespace,

en...,what ever, I will remove this option.

**Comment by hxsf** [comment]
reopen

**Comment by Glavin001** [comment]
Thanks for contributing!

**Comment by Glavin001** [comment]
Published to v0.30.0

**Pull request merged**